jax/ops/image/extract.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import numpy as onp
+
+from jax import lax
+from jax.ops.image import util
+from jax.numpy import lax_numpy as np
+
+
+def extract_image_patches(image, ksizes, strides, rates, padding):
+    # replicating tensorflow's implementation
+    num_dims = image.ndim
+    num_spatial_dims = num_dims - 2
+    data_format = "NHWC"
+
+    feature_dim = util.get_feature_dim(num_dims, data_format)
+
+    depth = image.shape[feature_dim]
+
+    kernel_shape = [ksizes[util.get_spatial_dim(num_dims, data_format, i)]
+                    for i in range(num_spatial_dims)]
+    kernel_size = onp.product(kernel_shape)
+    kernel_shape.append(1)
+    kernel_shape.append(kernel_size * depth)
+
+    iota_kernel_shape = (kernel_size, 1, kernel_size)
```

Comment:
Shouldn't this be `(kernel_size, depth, kernel_size)`?

Reply:
Yes! That's what I get for only testing on images with depth 1... Now fixed.